src/zarr/storage/_utils.py

```diff
@@ -74,11 +74,60 @@ def _join_paths(paths: Iterable[str]) -> str:
     """
     Filter out instances of '' and join the remaining strings with '/'.
 
-    Because the root node of a zarr hierarchy is represented by an empty string,
+    Parameters
+    ----------
+    paths : Iterable[str]
+
+    Returns
+    -------
+    str
+
+    Examples
+    --------
+    >>> _join_paths(["", "a", "b"])
+    'a/b'
+    >>> _join_paths(["a", "b", "c"])
+    'a/b/c'
     """
     return "/".join(filter(lambda v: v != "", paths))
 
 
+def _relativize_path(path: str, prefix: str) -> str:
+    """
+    Make a "\"-delimited path relative to some prefix. If the prefix is '', then the path is
+    returned as-is. Otherwise, the prefix is removed from the path as well as the separator
+    string "\".
+
+    If ``prefix`` is not the empty string and``path`` does not start with ``prefix``
```

Comment:
```suggestion
    If ``prefix`` is not the empty string and ``path`` does not start with ``prefix``
```

Reply:
And I guess since every string technically starts with the empty string, you can rephrase this as just "If path does not start with prefix ..."